### PR TITLE
Filters in password table

### PR DESCRIPTION
**Pull request by math07**
Done, was merged with math07:selectIndividualHashes (PR #54)

### CODE REVIEW

**Comment by math07** [comment]
For every cracked passwords #45, I put the username font in bold and I made the password background column dark green. 

Regarding #44, I wrote "EMPTY" for all "" passwords and put the password background column dark gray instead of green.

**Comment by shinnok** [comment]
I've checked out the branch and am looking at it. Observations:
1. There are too many colors now in it. Let's remove the alternateColorRows property on the table and use the alternate color only for the checked rows.
2. I'm not particularly fond of the green/grey columns for successfully cracked passwords. Not sure tuning out the color would suffice either. One option would be to remove the colors all together. For the empty password ones, center the text **EMPTY PASS** in the cell and apply the grey color to the text.

Some things are not working, like the filter not applying to the Formats column or the Show only selected checkbox, but then you didn't say you were finished with the work I think, so I'll hold on to testing that.

**Comment by math07** [comment]
> Some things are not working, like the filter not applying to the Formats column or the Show only selected checkbox, but then you didn't say you were finished with the work I think, so I'll hold on to testing that.

Yeah those things are not implemented since I'm waiting for your opinion regarding the performance problem I have on big files in #56 thread since maybe we'll no longer be able to use QSortFilterProxyModel depending on our decision in this discussion.

**Comment by math07** [comment]
I did what you said regarding #44 and #45 in c0be408 and I think that it looks WAY better like this.

**Comment by math07** [comment]
> I did what you said regarding #44 and #45 in c0be408 and I think that it looks WAY better now.

Oups, I put the "already cracked password" with the alternate color property instead of the checked one as you said. Fixed in a593b9e . Anyway, you can checkout both versions to see in action and see your preference.

**Comment by math07** [comment]
I implemented all the things needed(except the performance improvement for big files) and this small thing. Damn, I'm having so much troubles on a simple thing : when the show only checked hashes is checked, I can't hide rows as soon as the user uncheck them. 1)If I hide the row via the tableview method, the next filtering from QSortDataFilterModel won't work with that cell. 2) If I use QSOrtFilterProxyModel::removeRow(), it doesn't do anything. 3) There is no "invalide filter for this row only" in QSortFilterProxyModel so I have to invalidate the whole model to make it work, which is a no-brainer for big files. We don't want to redo the whole filtering everytime a user uncheck a row.

**Comment by shinnok** [comment]
I've checked the branch again, a couple amendments:
1. For the right click context menu, change the "Select all hashes" and deselect respectively to just "Include/Exclude in attack" and make them only check/uncheck the currently selected rows.
2. Invert the colors on the included/excluded rows. Primary for included, secondary for excluded. On OS X, that would be white for included, grey for excluded.
3. Just use invalidate filter, performance for 100k hashes is not that bad here by toggling "Show only" repeatedly for me. Is it really bad for you?

**Comment by shinnok** [comment]
We could also do with another checkbox, "Show only cracked", that would be very useful.

**Comment by math07** [comment]
> I've checked the branch again, a couple amendments:
> 1. For the right click context menu, change the "Select all hashes" and deselect respectively to just "Include/Exclude in attack" and make them only check/uncheck the currently selected rows.
> 2. Invert the colors on the included/excluded rows. Primary for included, secondary for excluded. On OS X, that would be white for included, grey for excluded.
> 3. Just use invalidate filter, performance for 100k hashes is not that bad here by toggling "Show only" repeatedly for me. Is it really bad for you?

I've done your changes. Regarding point 3, performance isn't that bad on 100k hashes files when toggling the show only checkbox. In 65df909, when the show only checked hashes is checked, I finally implemented auto hiding of rows as soon as the user uncheck them when the option is enabled in a way that doesn't perform bad. I'll do the "show only cracked" feature tomorrow.

File: forms/mainwindow.ui [hunk 1]
```diff
@@ -1473,6 +1580,16 @@
...
    </property>
   </action>
+  <action name="actionIncludeSelectedHashes">
+   <property name="text">
+    <string>Include in attack</string>
+   </property>
+  </action>
+  <action name="actionExcludeSelectedHashes">
+   <property name="text">
+    <string>Exclude in attack</string>
```
Comment (shinnok) [inline forms/mainwindow.ui]: Exclude from attack.

**Comment by math07** [comment]
All right Shinnok, I've implemented all the things that you told me in this PR and other threads #54 and #56. I'd say we are ready for a merge unless you have new comments.

**Pull request merged**